polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -229,50 +232,97 @@ pub fn worker_entrypoint(
 					},
 				};
 
-				// SAFETY: new process is spawned within a single threaded process. This invariant
-				// is enforced by tests.
-				let result = match unsafe { nix::unistd::fork() } {
-					Err(errno) => Err(error_from_errno("fork", errno)),
-					Ok(ForkResult::Child) => {
-						// Dropping the stream closes the underlying socket. We want to make sure
-						// that the sandboxed child can't get any kind of information from the
-						// outside world. The only IPC it should be able to do is sending its
-						// response over the pipe.
-						drop(stream);
-						// Drop the read end so we don't have too many FDs open.
-						drop(pipe_reader);
-
-						handle_child_process(
-							pvf,
-							pipe_writer,
-							preparation_timeout,
-							prepare_job_kind,
-							executor_params,
-						)
-					},
-					Ok(ForkResult::Parent { child }) => {
-						// the read end will wait until all write ends have been closed,
-						// this drop is necessary to avoid deadlock
-						drop(pipe_writer);
-
-						handle_parent_process(
-							pipe_reader,
-							child,
-							temp_artifact_dest.clone(),
-							worker_pid,
-							usage_before,
-							preparation_timeout,
-						)
-					},
-				};
-
-				gum::trace!(
-					target: LOG_TARGET,
-					%worker_pid,
-					"worker: sending result to host: {:?}",
-					result
-				);
-				send_response(&mut stream, result)?;
+				cfg_if::cfg_if! {
+					if #[cfg(any(target_os = "linux"))
+					] {
+						let stack = 1024 * 1024;
+						let mut stack: Vec<u8> = vec![0u8; stack];
+						let flags = CloneFlags::CLONE_NEWCGROUP
+						| CloneFlags::CLONE_NEWIPC
+						| CloneFlags::CLONE_NEWNET
+						| CloneFlags::CLONE_NEWNS
+						| CloneFlags::CLONE_NEWPID
+						| CloneFlags::CLONE_NEWUSER
+						| CloneFlags::CLONE_NEWUTS
+						| CloneFlags::from_bits_retain(libc::SIGCHLD);
+
+						// SAFETY: new process is spawned within a single threaded process. This invariant
+						// is enforced by tests. Stack size being specified to ensure child doesn't overflow
+						let result = match unsafe { nix::sched::clone(
+							Box::new(|| {
```

Comment:
Here I couldn't find a way to close the pipe reader and the UNIX stream. Not sure if it is needed.

Reply:
Why is it, do you get an error? I believe dropping either of those is not strictly needed, but would be nice to do.